pandas/tests/io/json/test_json_table_schema.py

```diff
@@ -86,6 +87,16 @@ def test_multiindex(self):
         assert result == expected
 
 
+class TestParseSchema(object):
+
+    def test_empty_json_data(self):
+        # GH21287
+        df = pd.DataFrame([], columns=['a', 'b', 'c'])
+        json = df.to_json(None, orient='table')
+        df = parse_table_schema(json, True)
+        assert df.empty
```

Comment:
To make sure that we've preserved the frame metadata we should use the `tm.assert_frame_equal` function here. Typically with tests we will:

1. Create a variable called `expected`, which is very explicit about what we want (here that's just a copy of `df`)
2. Assign to a variable called `result`, which here would be the result of `pd.read_json`
3. Make sure result and expected match using `tm.assert_frame_equal`